src/client/src/components/pages/courses/courseDetails/index.jsx

```diff
@@ -7,26 +7,33 @@ import {
 import Form from "./../../../abstract/Form";
 import Footer from '../../../abstract/footer';
 import axios from "axios";
+import contextHoc from './../../../abstract/HOC/contextHoc';
 
-export default class index extends Component {
+class index extends Component {
   state = initialState;
 
   onChange = event => {
     const { value, name } = event.target;
     this.setState({ [name]: value });
   };
 
-  getData = async () => {
+  getDetails = async () => {
+    const { dispatch } = this.props.context;
     const id = this.props.match.params.id;
-    const data = await axios(`/api/v2/course/${id}`);
-    const details = data.data.details;
-    const startDate = details.course_start.split("T")[0];
-    const endDate = details.course_end.split("T")[0];
-    this.setState({...details, course_start:startDate, course_end:endDate});
+    axios(`/api/v2/course/${id}`).then(result => {
+
+      const { data } = result;
+      const startDate = data.course_start.split("T")[0];
+      const endDate = data.course_end.split("T")[0];
+      this.setState({ ...data, course_start:startDate, course_end:endDate });
+    }).catch(error => {
+      dispatch({ type: 'ERROR_PAGE', payload: { ErrorPage: error.response.status } })
```

Comment:
On our projects, we usually put our action names into consts and then import them when dispatching.  So for example, in an actions.js file:
`const ERROR_PAGE = 'ERROR_PAGE'`,
and then in this file and the other component files:
```
import { ERROR_PAGE } from '../actions.js'
...
dispatch({ type: ERROR_PAGE, ... })
```
it's not a required change and might seem unnecessary, but it helps with debugging typos (if you typo with the imported constants, it will be a lot more obvious what went wrong than if you typo one one of these string literals like you have here).